### DISCUSSION TITLE
single-line mode v.s. multi-line mode

### DISCUSSION THREAD

**Discussion by xixiaofinland** [xixiaofinland/afmt#23] [General]
When the code exceeds max_width (usually 80, 100 or 120 by default), some types of nodes should be split into multi-lines. 
A typical example is parameters in the method declaration.

There are several settings in Prettier Apex (PA) that I don't think should be the default in Afmt.

**Comment by xixiaofinland** [comment]
1. In PA, class declaration won't split when it exceeds max_width, 
```
public with sharing class VerrlongNameClassAbcIsHereYouCanSeeAlright implements Database.Batchable<SObject> {}
```

I think it should split before `implements`
```
public with sharing class VerrlongNameClassAbcIsHereYouCanSeeAlright 
    implements Database.Batchable<SObject> {}
```

**Reply by aheber** [reply]
I would prefer to wrap items like that rather than exceed line width. Similar with things like `extends ExtendedClass`.

**Comment by xixiaofinland** [comment]
2. In PA, Enum is always split regardless of the length. For example, `public enum En { X, Y }`

In PA, it's formatted into
```
public enum En {
  X,
  Y
}
```

I think it should be split only when the length > max_width, i.e. the same rule as in the method declaration.

**Reply by aheber** [reply]
This is completely my preference. I would prefer it all on the same line unless it has a large number of elements. Even if the sum width of those elements are within the print boundary. Something like 4 elements in a line is OK but 5 I think I would prefer a print. (That is a mostly made up number but just giving an example)

I'd be OK if this just followed line width rules like anything else though.

**Comment by xixiaofinland** [comment]
3. Shall different `get {...}` `set {...}` be in a single line when it's shorter than max_width?

```
public class Me {
  public integer prop {
    get;
    set { prop = value; }   // <-- I bet single line is better when it's short?
  }
}
```

PA will format the `{...}` part regardless the length
```
public class Me {
  public integer prop {
    get;
    set {
      prop = value;
    }
  }
}
```

**Reply by aheber** [reply]
I'd like to always have `{ get; set; }` on the same line, even if it exceeds line width (I think)

I don't mind example 1 when it is reasonable but i would be made if it was always example 2 as soon as either of them have a block statement attached. I don't know that I would see it as worth the complexity/logic to go out of my way to give it the variation.
